pkg/sql/sem/tree/create.go

```diff
@@ -2160,3 +2160,34 @@ func (node *CreateExtension) Format(ctx *FmtCtx) {
 	// users attempt to load.
 	ctx.WriteString(node.Name)
 }
+
+// ConnectionLabelSpec describes the labeling specification for an external
+// connection.
+type ConnectionLabelSpec struct {
```

Comment:
Can we use a generic component here? Or is per-node more common?

Reply:
I thought about it, I've introduced a new `labelSpec` that is now used by Schedules and External Connections. Note, for schedules labels are optional, but for external connections they're not so we still need a `schedule_label_spec` rule.